posix/posix.c

```diff
@@ -477,6 +477,77 @@ static int posix_create(const char *filename, int type, mode_t mode, oid_t dev,
 	return err;
 }
 
+int posix_statvfs(const char *path, int fildes, struct statvfs *stat)
+{
+	oid_t oid, dev;
+	oid_t *oidp, *devp;
+	open_file_t *f;
+	msg_t msg;
+	int err = EOK;
+
+	if (((path == NULL) && (fildes < 0)) ||
+			((path != NULL) && (fildes != -1))) {
+		return -EINVAL;
+	}
+
+	if (path == NULL) {
+		err = posix_getOpenFile(fildes, &f);
+		if (err < 0) {
+			return err;
+		}
+		oidp = &f->oid;
+		devp = NULL;
+	}
+	else {
+		if (proc_lookup(path, &oid, &dev) < 0) {
+			return -ENOENT;
+		}
+		oidp = &oid;
+		devp = &dev;
+	}
+
+	/* Detect mountpoint */
+	if ((devp != NULL) && (oidp->port != devp->port)) {
+		hal_memset(&msg, 0, sizeof(msg));
```

Comment:
Just being curious: why did you replace a struct initializer with memset on msg?

Reply:
struct initializer could cause GCC to generate `memset` which is not available in kernel and plo

Reply:
IMHO we should conform to the GCC requirements in freestanding environment (by providing symbol aliases or wrappers if necessary), I see that since `C23` even the standard requires we provide `memset`/`memcpy`.

Obviously to be done outside of the scope of this PR.